aten/src/ATen/native/quantized/cpu/quant_utils.h

```diff
@@ -5,6 +5,35 @@
 #include <cmath>
 
 namespace quant_utils {
+namespace {
+  float RawUint16ToFp16(unsigned short value) {
+    // Convert raw 16 bits half precision floating point number
+    // to single precision floating point number.
+    const unsigned short sign_bits = value >> 15;
+    const unsigned short exponent_bits = value >> 10 & 0x1f;
+    const unsigned short significand_bits = value & 0x3ff;
+
+    const float sign = sign_bits ? -1 : 1;
+    const float significand =
+        1 + significand_bits * 0.0009765625f; // 0.0009765625f = 0x1p-10 = 2^-10;
+    const float exponent = exponent_bits - 0xf;
+
+    return sign * std::ldexp(significand, exponent);
+}
+
+template <typename T>
+bool CheckAndSaturate(T max_val, T* element) {
```

Comment:
optional: I realize this is just moving code around, but this API seems to require careful reading to understand it correctly.  Might be worth a refactor in the future to make it a void return type, and have the status be a var passed by reference.